### PR TITLE
Closes AgileVentures/MetPlus_tracker#127

### PR DESCRIPTION
**Pull request by patmbolger**
Added agencies_controller with two actions: edit and update
Added spec controller tests
Added cucumber integration tests

### CODE REVIEW

File: app/helpers/agency_admin_helper.rb [hunk 1]
```diff
@@ -4,4 +4,8 @@ def current_user_agency_admin?
     return false unless ((person = current_user.actable).is_a? AgencyPerson)
     person.agency_roles.pluck(:role).include? AgencyRole::ROLE[:AA]
   end
+  
+  def person_full_name(person)
```
Comment (joaopapereira) [inline app/helpers/agency_admin_helper.rb]: Shouldn't this be a model method?

File: app/models/agency.rb [hunk 2]
```diff
@@ -9,10 +9,7 @@ class Agency < ActiveRecord::Base
   validates :phone, :phone => true
   validates :email, :email => true
   validates :website, :website => true
-  validates :fax, :phone => true
```
Comment (joaopapereira) [inline app/models/agency.rb]: Why do we stop validating the fax?

File: config/routes.rb [hunk 3]
```diff
@@ -1,24 +1,22 @@
...
-
-#  post '/login' => 'session#create'
-
-#  get '/logout' => 'session#destroy'
+  
+  resources :agencies, only: [:edit, :update]
+  
+  root 'main#index'
+  
+  get 'agency_admin/home'
```
Comment (joaopapereira) [inline config/routes.rb]: Maybe this URL should look like /admin/agency/
Like this is more rest like

**Comment by joaopapereira** [comment]
Looks good. Just clarify my comments

**Comment by patmbolger** [comment]
I moved the 'full_name' method to User as this will be needed by both AgencyPerson and CompanyPerson.

I moved the agencies controller under the 'admin' namespace.

File: app/models/user.rb [hunk 4]
```diff
@@ -37,5 +37,9 @@ def self.is_company_contact?(user)
       return false unless user.actable_type == "CompanyPerson"
       user.actable.company_roles.pluck(:role).include? CompanyRole::ROLE[:EC]
     end
+    
+    def self.full_name(person)
```
Comment (joaopapereira) [inline app/models/user.rb]: This should be a instance method not a class method

File: config/routes.rb [hunk 5]
```diff
@@ -1,24 +1,22 @@
...
-
-#  post '/login' => 'session#create'
-
-#  get '/logout' => 'session#destroy'
+  
+  resources :agencies, path: '/admin/agencies', only: [:edit, :update]
+  
+  root 'main#index'
+  
+  get 'agency_admin/home'
```
Comment (joaopapereira) [inline config/routes.rb]: Shouldn't this rout be under the resources part above?

File: app/models/agency.rb [hunk 6]
```diff
@@ -9,10 +9,7 @@ class Agency < ActiveRecord::Base
...
-  
-  @@agency_admin = nil
```
Comment (joaopapereira) [inline app/models/agency.rb]: I believe we have one Issue for these changes, One to refactor this part?
Or are you thinking in the refactor ticket to refector the find_user_with_role function?

File: app/models/user.rb (hunk 4 above)
Comment (patmbolger) [inline app/models/user.rb]: what was I thinking?  thanks for catching that ....

File: app/models/agency.rb (hunk 6 above)
Comment (patmbolger) [inline app/models/agency.rb]: Yes, I have refactored that in the next (pending ) PR.

File: config/routes.rb (hunk 5 above)
Comment (patmbolger) [inline config/routes.rb]: I moved it under admin.

**Comment by joaopapereira** [comment]
Looks good.

**Pull request merged**